### PR TITLE
chore: use https protocol and update devfile scheme to 2.2.2 for dotnetcore31 and angular

### PR DESCRIPTION
**Pull request by vitaliy-guliy**
### What does this PR do?:
_Summarize the changes. Are any stacks or samples added or updated?_

Updated `dotnetcore31` and `nodejs-angular` devfiles.

Linked with https://github.com/devfile-samples/devfile-stack-nodejs-angular/pull/3

.NET Core 3.1

Screenshot from 2024-05-15 10-25-49

Angular 2.0.2, 2.1.0, 2.2.0

Screenshot from 2024-05-15 10-41-49

### Which issue(s) this PR fixes:
_Link to github issue(s)_
Part of https://github.com/eclipse-che/che/issues/22869

### PR acceptance criteria:

- [ ] Contributing guide
_Have you read the [devfile registry contributing guide](https://github.com/devfile/registry/blob/main/CONTRIBUTING.md) and followed its instructions?_
- [ ] Test automation
_Does this repository's tests pass with your changes?_
- [ ] Documentation
_Does any documentation need to be updated with your changes?_
- [ ] Check Tools Provider
_Have you tested the changes with existing tools, i.e. Odo, Che, Console? (See [devfile registry contributing guide](https://github.com/devfile/registry/blob/main/CONTRIBUTING.md) on how to test changes)_

### How to test changes / Special notes to the reviewer:

### CODE REVIEW

**Review by svor — COMMENTED** [review]
About nodejs-angular stack, you don't need to make changes for all devfiles. Just create a new folder 2.2.2 and put into it a devfile with your changes, then add this version into [stack.yaml](https://github.com/devfile/registry/blob/main/stacks/nodejs-angular/stack.yaml#L12) file and mark it as a default

**Review by thepetk — COMMENTED** [review]
As mentioned by @svor I think better not to apply the same change to all the `nodejs-angular` versions. Creating a new default (`2.2.1`) applying all updates there I think is sufficient.

File: stacks/nodejs-angular/2.1.0/devfile.yaml [hunk 1]
```diff
@@ -25,26 +25,27 @@ components:
   - container:
       endpoints:
         - name: http-angular
```
Comment (ibuziuk) [inline stacks/nodejs-angular/2.1.0/devfile.yaml:27]:
```suggestion
        - name: https-angular
```

File: stacks/dotnetcore31/devfile.yaml [hunk 2]
```diff
@@ -35,7 +35,8 @@ components:
         - name: ASPNETCORE_URLS
           value: http://*:8080
       endpoints:
-        - name: http-dotnetcore
+        - name: https-dotnetcore
```
Comment (svor) [inline stacks/dotnetcore31/devfile.yaml:38]: trying to start dotnet devfile in che:
```
Failed to fetch resources. Devfile schema validation failed. Error: 0: instance.components[0].container.endpoints[0].name does not meet maximum length of 15
```
screenshot-che-dogfooding apps che-dev x6e0 p1 openshiftapps com-2024 05 17-16_27_52

the same error in the PR check build

Maybe replace the endpoint's name with **https-dotnet** or just **dotnetcore**

Comment (vitaliy-guliy) [reply stacks/dotnetcore31/devfile.yaml]: done
**https-dotnet** used

**Review by svor — APPROVED** [review]
lgtm

File: stacks/dotnetcore31/devfile.yaml [hunk 3]
```diff
@@ -35,7 +35,8 @@ components:
         - name: ASPNETCORE_URLS
           value: http://*:8080
       endpoints:
-        - name: http-dotnetcore
+        - name: https-dotnet
```
Comment (thepetk) [inline stacks/dotnetcore31/devfile.yaml:38]: Sound better to me if it remains `https-dotnetcore`. WDYT?

**Review by thepetk — COMMENTED** [review]
Tested locally and looks good to me. I left only a minor comment about the dotnetcore endpoint.

@kadel as you are the owner of the `dotnetcore31` stack could you please review the PR?

File: stacks/dotnetcore31/devfile.yaml (hunk 3 above)
Comment (thepetk) [reply stacks/dotnetcore31/devfile.yaml:38]: Oh I see! Then I think is fine to leave it as is

Comment (vitaliy-guliy) [reply stacks/dotnetcore31/devfile.yaml:38]: Endpoint name `https-dotnetcore` appears to be too long. 
We have already discussed it here https://github.com/devfile/registry/pull/398#discussion_r1605008509

Anyway, `https-dotnetcore` much better suits here, I wonder whether it possible to increase the maximum length of the name somewhere?

Comment (thepetk) [reply stacks/dotnetcore31/devfile.yaml:38]: Is mentioned [here](https://devfile.io/devfile-schemas/2.2.2.json) in the schema. IMO is better if we keep the current name and don't change the devfile schema.

**Review by ibuziuk — APPROVED** [review]

**Pull request merged**